### ISSUE TITLE
AE Poles Web Map / App - Harrell M.

### ISSUE THREAD

**Issue by jaime-mckeown** [cityofaustin/atd-data-tech#4304]
Create a Web Map / App for Harrell's mapping processes to get him out of ArcMap. Notes from meeting with him are located in #4219.

**Comment by jaime-mckeown** [comment]
- Created Web Map and Web App. Added the ability to Filter, Draw, and Print.
- The Print Widget will not print the basemap unless an Esri Basemap is turned on or aerials. I talked to Allison and PARD about this problem as she has already done some research into the issues with the Print Widget.

**Comment by jaime-mckeown** [comment]
- turn off fiber optic cable
- returning to last map page when opening
- removed the AE - Poles Class layer and filter as it's not useful to Harrell.
- Sent him a link to the Web App so he can start using it.

**Comment by jaime-mckeown** [comment]
- [ ] Create new Mxds from scratch and email to Harrell.

**Comment by jaime-mckeown** [comment]
Web Map requests:

> I really like what I am seeing on this map, very user friendly! Having said that, I have a few questions.
> 1.	Can the street light pole numbers be added with the option to turn them on and off?
> **Yes, I can add the streetlight pole numbers.**
> 2.	Can I add poles to the map? This is very import. I would like to add poles to the map that belong to ATD and other entities. There are poles that belong to TWC and AT&T that ATD’s fiber are attached to but these poles are not shown on the AE GIS pole layer.
> **To keep the integrity of the AE Poles layer, how about I create a new “Other Poles” layer that has the same schema that you can add utility poles to and edit?**
> 3.	Can the pole indicators be made visible when the map is zoomed out? Using the scale bar at the bottom left hand corner, the pole indicators come up at 0 – 600ft. Can they be made visible at 0 - .3 mi or 0 - .6 mi? I prefer 0 - .6 mi.
> **Yes, I can change the visibility of the poles.**
> 4.	Can the pole numbers be made a little bit bigger? I know it will be messy when the map is zoomed out, but when I zoom in to say 0 – 100 ft or 0 – 200 ft if they were a little bit bigger it would nice. I guess what I’m asking is, can they be scaled with the map zoom. Ex. The more I zoom in, can the numbers get bigger with the zoom?
> **Yes, they can be made bigger. I will see if I can set a reference scale so they change size, but I’m not sure if that’s possible in ArcGIS Online.**
> 5.	Can a measuring tool be added so I can measure between poles?
> **Yes, I can add the Measure tool, it will be added to the toolbar at the top right of the map**
> 6.	Can a field be added to the pole info box where I can add notes or info about that pole? For example, add the pole number if AE doesn’t show an asset id number, what com assignment we are on the pole or who owns the pole. The more I think about it, this would be very useful. Maybe we could call it “ Notes “ or something like that.
> **Yes, I can definitely add a Notes field. I can also make the asset_id field editable so you can add missing pole numbers.**

**Comment by jaime-mckeown** [comment]
To Harrell:

> As I’m thinking about making the AE Poles layer editable and adding a Notes field, I’m realizing that this will not work. Since it is overwritten by the updated data that I receive from AE multiple times a year, all of your edits will be lost when the data is updated. With that being said, if I create a new layer specific for your needs where you can add AE Poles, other utility poles, COA poles and notes, it will become your authoritative dataset that will never be overwritten. As each new version of AE Poles is added, you will be able to modify your copy based on the updates that come through.
> 
> It may be more work in the beginning as this layer will be empty, but it will grow over time as you add more and more data.

From Harrell:

> RATS, that’s not the answer I was wanting. If I’m reading this correctly, even if you put a separate notes box in this map for each pole that we could turn on and off, every time you get the new data from AE, we would lose the data in the notes box also.
> I may not be understanding what you mean. Can we talk about this sometime tomorrow on teams?

**Comment by jaime-mckeown** [comment]
Sent questions to Blake:

> I have a couple of questions for you regarding the AE Utility Poles GIS data that have come up while I’ve been working with ATD’s Arterial Management staff that use the data.
> -	They have noticed that there are many poles with blank asset_id fields. According to our users, they don’t normally see blank asset_id fields. Has there been a change with the data that has affected the asset_id field?
> -	Which field is your actual unique ID field, asset_id or id? We are interested in creating a related table that is referencing the AE Poles data, but I want to make sure that the relationships will not be lost over time.

**Comment by jaime-mckeown** [comment]
Blake's response:

> There were a couple of poles missing numbers by mistake, I’m getting them corrected.  But the vast majority of them are pre-construction.  When a job gets sent to Construction we put the job in GIS so the control center will be able to see what will be coming in when it’s done.  Poles don’t get numbers until the job is completed.  We’ve been doing this for a couple years so they should have seen some poles without numbers before, but there are quite a few in the system now so they might have only noticed them recently.  In our system there is an attribute field to identify these objects.  Since we so rarely share our data I guess we never thought about how these would come through in the export.  I’ll try to remember to add that attribute field next time I run the export.
> Both numbers should be unique.  The asset_id field is the unique number that is physically on the pole in the field.  The id field is the unique number for the object in the database.  For your use relating a table I would use the id field.  The tag on a pole can be replaced.  In that case, we would update the asset_id but the id would stay the same.

**Comment by jaime-mckeown** [comment]
- Let Harrell know what Blake said. Checked with Harrell regarding the fields he would like to add to the new table. I have configured what he has gave me, but still waiting for all possibilities for Over Lash field.
- Created Domains for all fields that can have them to prevent mis-typing.
- Created new table and loaded all records from the AE Poles layer, only including the id field from the layer.
- Published to AGOL, but it looks like the relationship has to be created in Pro.
- Double checked that the id field was actually unique, there are no duplicate id fields
- Created a relationship class between the current AE Poles and AE Poles AMD Notes table.
- Each time I receive an updated version of the AE Poles data I will need to export the current feature layer to set up a new relationship class between the new AE Poles data and the AMD Notes table that is being updated in AGOL.

**Comment by jaime-mckeown** [comment]
- The related table will not work as an option as I don't want the actual feature layer to be edited. It was complicated to me, so I scrapped that option.
- The easiest option was to add Harrell's requested fields to the AE Poles layer. When the feature layer was re-published, I made it Editable, but only Attributes, no Geometry. Then in the pop-up I only allowed the new fields to be editable. This will make editing simpler for Harrell to figure out as now he just needs to click on a point, click the 3 dots, Edit, then the Edit Window pops up for editing. All new fields have domains setup except one which needs to be a free form text field. 
- Emailed Harrell directions with screenshots so he can try it out.
